website/pages/chapters/ridge.js

```diff
@@ -0,0 +1,845 @@
+        </p>
+        <ul>
+          <li>
+            Change the quality metric to not include the intercept in the
+            regularization term. Define{" "}
+            <IM math={`w_{rest} = [w_1, ..., w_D]`} /> to not include the
+            intercept <IM math={`w_0`} />.
+            <BM
+              math={`\\hat{w_0}, \\hat{w}_{rest} = \\min_{w_0, w_{rest}} RSS(w_0, w_{rest}) + \\lambda \\left|\\left|w_{rest} \\right| \\right|_2^2`}
+            />
+            Our predictor will then be in terms of these values.
+            <BM
+              math={`\\hat{y} = \\hat{w_0} + \\sum_{j=1}^D \\hat{w}_{rest,\\ j} \\cdot x`}
+            />
+          </li>
+          <li>
+            Center the data so the <IM math={`y`} /> values are have mean 0. If
+            this is the case, then favoring a <IM math={`w_0`} /> is not a
+            problem since the data is centered at 0.
+          </li>
+        </ul>
+
+        <p>
+          The other big problem we run into is the effect that the scale of the
+          features has on the coefficients. We discussed earlier in this chapter
+          that features on a smaller scale might require larger coefficients to
+          compensate. This can lead to a false penalty for a coefficient that
+          needs to be large just to deal with such features. In other words,
+          it's not always the case that large coefficients mean overfitting, it
+          could also mean there is truly a large increase in output per unit of
+          the input due to the units.
+        </p>
+
+        <p>
+          A common fix to this problem is to <b>normalize</b> the data so that
+          all of the features have mean 0 and standard deviation 1. This
+          prevents having to deal with different features on entirely different
+          range since you scale them down to the same range.
+        </p>
+
+        <p>
+          There are a lots of ways to define how to normalize your data. A
+          common one is as follows.
+        </p>
+
+        <BM
+          math={`\\tilde{h}_j(x_i) = \\frac{h_j(x_i) - \\mu_j(Train)}{\\sigma_j(Train)}`}
+        />
+
+        <p>
+          Where we define <IM math={`\\mu_j(Train)`} /> to be the mean of
+          feature <IM math={`j`} /> in the training set.
+        </p>
+        <BM
+          math={`\\mu_j(Train) = \\frac{1}{|Train|} \\sum_{x \\in Train} h_j(x)`}
+        />
+        <p>
+          We define <IM math={`\\sigma_j(Train)`} /> to be the standard
+          deviation of feature <IM math={`j`} /> in the training set.{" "}
+        </p>
+        <BM
+          math={`\\sigma_j(Train) = \\sqrt{\\frac{1}{|Train|} \\sum_{x\\in Train}\\left( h_j(x) - \\mu_j(Train) \\right)^2}`}
+        />
+
+        <p>
+          So by pre-processing our data to be normalized, we don't have to worry
+          about features with wildly different scales being falsely penalized by
+          the regularization term.
+        </p>
```

Comment:
Maybe add a practice problem of doing normalization. i know when I was learning this stuff, I felt like regularization was a fuzzy concept. IDK the best way to do it, but maybe make them calculated the normalization by hand, and give them a learned model, so they can see the cofficient values? ALSO while you don't normalize y (it looks like) I feel like people not only normalize x, but y as well?

Reply:
I think a practice problem for regularization is a great idea that I'll probably save for class. I have left out most of the poll everywhere questions from the slides because I want to use those for in-person activities. I have to have SOMETHING to do in class 😛 

I added a note about normalizing y's. It is common in practice, but not strictly necessary. The big problem with the inputs is having them on _different_ scales. If you made them all mean 50 std deviation 47, it would be fine. The problem comes from having one feature that requires large coefficients and one that requires small since then you don't penalize them the same.